src/Codec/Xlsx/Parser/Stream.hs

```diff
@@ -264,6 +287,20 @@ instance HasPSFiles SheetState where
 instance HasPSFiles SharedStringState where
   fileLens = ss_file
 
+--
+-- Here begins the types for the part of this module that uses the "zip" library parser ---
+--
+-- TODO: decide whether to remove one or the other approaches, or unify them, or separate modules
+data XlsxMState = MkXlsxMState
+  { _xs_shared_strings :: IORef (Maybe (Map Int Text))
+--  , _xs_sheet_names :: IORef (Maybe (Map Int Text))
+  }
+makeLenses 'MkXlsxMState
+
+newtype XlsxM a = XlsxM { unXlsxM :: ReaderT XlsxMState Zip.ZipArchive a }
```

Comment:
In my first Haskell position I was told to re-write something that I'd written with StateT to use 'ReaderT over IO/STRef' instead. There are some reasons to do so given here: https://www.fpcomplete.com/blog/2017/06/readert-design-pattern/ Those opinions lead to unliftio and sticking to monads that don't involve StateT (specifically to avoid the complexities of monad-control further down the line, I think). So from my perspective it's more in culture with our proprietary base. I don't have strong opinions either way on this.

So I was trying to follow past experience, best practices, and "wanted to make it fast". Again, I don't have that strong opinions here though.

Reply:
I'm familiar with the blogpost[1], but I didn't realize this was also about stateT, I thought just writerT.

In the caveats sections, snoyman mentions stateT is okay for mostly pure code that's not parallelizable or doesn't need state to recover from errors/report errors. I suspect this parsing fits that description. 

In any case I think it's also good to leave it like it is now I realize it's part of the readerT pattern. (not to imply I thought it was not good in the first place, I just thought it was peculiar, now I don't even think it peculiar anymore).

[1]: I originally based most of my mtl understanding around that purification example!

Reply:
I think you're right that StateT would be fine, and nicer too. The other parts of the Parser.Stream module also use StateT, and even `ZipArchive` itself is just a StateT over IO. When I come back to this PR in a day or two I'll change it to use StateT (small change).